modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/IgniteH2Indexing.java

```diff
@@ -642,7 +648,7 @@ private void removeTable(TableDescriptor tbl) throws IgniteCheckedException {
         if (log.isDebugEnabled())
             log.debug("Removing query index table: " + tbl.fullTableName());
 
-        Connection c = connectionForThread(tbl.schema());
```

Comment:
It's the consequences of doing IgniteH2Indexing#schema method non-static. This approach allows us not to create any extra fields in TableDescriptor class: like String schemaName for ex.